rpcserver.go

```diff
@@ -6591,14 +6592,18 @@ func marshalPeerAcceptedBuyQuotes(
 		[]*rfqrpc.PeerAcceptedBuyQuote, 0, len(quotes),
 	)
 	for scid, quote := range quotes {
+		rpcAskAssetRate := &rfqrpc.FixedPoint{
+			Coefficient: quote.AssetRate.Coefficient.ToUint64(),
```

Comment:
With sufficient scale (this value is already multiplied by the scale), then this won't always fit into a uint64. 

We should be using the main constructor when ever possible to create these integers, using a base value (like in dollars/cents), then scaling up from there: https://github.com/lightninglabs/taproot-assets/blob/65539b6fd8e11657f96353a87714ed2d3e4399a6/rfqmath/fixed_point.go#L159-L169

We want ppl to think in the normal values, adding the scale to basically normalize things, not think in the largest possible number representation.

Reply:
I think we should add something to minimise the coefficient and scale of a fixed-point before trying to export as `uint64`:

```golang
// countTrailingZeros returns the number of trailing zeros in the given integer.
func countTrailingZeros[T Int[T]](v T) uint8 {
	var (
		val = v

		// count will be the number of trailing zeros in the given
		// integer.
		count = uint8(0)
	)

	// We'll use these constants to avoid creating new instances
	// unnecessarily.
	var (
		zero = NewInt[T]().FromUint64(0)
		ten  = NewInt[T]().FromUint64(10)
	)

	// Count the number of trailing zeros in the given integer `val`.
	// The counting process continues as long as `val` is greater than zero
	// and dividing `val` by 10 does not result in a loss of precision.
	// This means that dividing by 10 successfully removes a trailing zero
	// from the number.
	for val.Gt(zero) && val.Div(ten).Mul(ten).Equals(val) {
		val = val.Div(ten)
		count++
	}

	return count
}

// MinParams returns a new FixedPoint with the smallest possible scale
// and coefficient, preserving precision. Useful for converting a fixed-point
// coefficient to a `uint64` representation.
func (f FixedPoint[T]) MinParams() FixedPoint[T] {
	trailingZeros := countTrailingZeros(f.Coefficient)

	// Determine the new scale based on the number of trailing zeros in
	// the coefficient.
	var newScale uint8
	if trailingZeros > f.Scale {
		// If the number of trailing zeros is greater than the current
		// scale, we can effectively use a minimal scale of 0.
		newScale = 0
	} else {
		// Otherwise we can only scale down to a new scale that is the
		// difference between the current scale and the number of
		// trailing zeros. Scaling down further will result in a loss
		// of precision in the coefficient.
		newScale = f.Scale - trailingZeros
	}

	// Return a new FixedPoint scaled to the smallest possible scale.
	return f.ScaleTo(newScale)
}
```

Reply:
In the latest commits, when marshalling into RPC, I've changed `quote.AssetRate.Coefficient.ToUint64()` into `ToUint64Safe`.